### PR TITLE
Documenting NSX-T support

### PR DESCRIPTION
**Pull request by stuclem**
Some content reorg and fixes https://github.com/vmware/vic-product/issues/1088

### CODE REVIEW

**Comment by stuclem** [comment]
I have updated all network-related docs to state that NSX-T logical switches can be used as well as vDS port groups. 

@yuyangbj @lgayatri @jitinkumar2018 can you please review this PR? Thanks!

File: docs/user_doc/vic_vsphere_admin/bridge_network.md [hunk 1]
```diff
@@ -10,40 +10,44 @@ In Docker terminology, the bridge network on a VCH corresponds to the default br
...
+Before you deploy a VCH, you must create a VMware vSphere Distributed Switch and a port group, an NSX Data Center for vSphere logical switch, or an NSX-T Data Center logical switch for the bridge network. For information about how to create a port group or logical switch, see [Networking Requirements for VCH Deployment](network_reqs.md#vchnetworkreqs).
 
 **IMPORTANT** 
 
-- Do not specify the same port group as the bridge network for multiple VCHs. Sharing a port group between VCHs might result in multiple container VMs being assigned the same IP address. 
-- Do not use the bridge network port group as the target for any of the other VCH networking options.
-- Do not use the bridge network for any other VM workloads.
+- You must create a dedicated vSphere port group, or an NSX Datacenter for vSphere logical switch, or an NSX-T Data Center logical switch for the bridge network for every VCH. Do not specify the same interface as the bridge network for multiple VCHs. Sharing an interface between VCHs might result in multiple container VMs being assigned the same IP address.
```
Comment (yuyangbj) [inline docs/user_doc/vic_vsphere_admin/bridge_network.md]: "Do not specify the same interface" --- using interface here is strange, could you please use port group or logical switch?

File: docs/user_doc/vic_vsphere_admin/bridge_network.md [hunk 2]
```diff
@@ -10,40 +10,44 @@ In Docker terminology, the bridge network on a VCH corresponds to the default br
...
+Before you deploy a VCH, you must create a VMware vSphere Distributed Switch and a port group, an NSX Data Center for vSphere logical switch, or an NSX-T Data Center logical switch for the bridge network. For information about how to create a port group or logical switch, see [Networking Requirements for VCH Deployment](network_reqs.md#vchnetworkreqs).
 
 **IMPORTANT** 
 
-- Do not specify the same port group as the bridge network for multiple VCHs. Sharing a port group between VCHs might result in multiple container VMs being assigned the same IP address. 
-- Do not use the bridge network port group as the target for any of the other VCH networking options.
-- Do not use the bridge network for any other VM workloads.
+- You must create a dedicated vSphere port group, or an NSX Datacenter for vSphere logical switch, or an NSX-T Data Center logical switch for the bridge network for every VCH. Do not specify the same interface as the bridge network for multiple VCHs. Sharing an interface between VCHs might result in multiple container VMs being assigned the same IP address. 
+- Do not use the bridge network interface as the target for any of the other VCH networking options.
```
Comment (yuyangbj) [inline docs/user_doc/vic_vsphere_admin/bridge_network.md]: I do not understand here, do you mean we cannot use the word "bridge"??

File: docs/user_doc/vic_vsphere_admin/bridge_network.md [hunk 3]
```diff
@@ -10,40 +10,44 @@ In Docker terminology, the bridge network on a VCH corresponds to the default br
...
-
+The `--bridge-network` option is **optional** if you are deploying a VCH to an ESXi host that is not managed by vCenter Server. In this case, if you do not specify `--bridge-network`, `vic-machine` creates a vSphere Distributed Switch and a port group that each have the same name as the VCH. You can optionally specify this option to assign an existing port group or logical switch for use as the bridge network for container VMs. You can also optionally specify this option to create a new port group that has a different name to the VCH.
 
-<pre>--bridge-network <i>port_group_name</i></pre>
+<pre>--bridge-network <i>port_group_or_logical_switch_name</i></pre>
 
-If you do not specify `--bridge-network` or if you specify an invalid port group name, `vic-machine create` fails and suggests valid port groups. 
+If you do not specify `--bridge-network` or if you specify an invalid port group or logical switch name, `vic-machine create` fails and suggests valid port groups or or logical switches.
```
Comment (yuyangbj) [inline docs/user_doc/vic_vsphere_admin/bridge_network.md]: two "or" here.

File: docs/user_doc/vic_vsphere_admin/bridge_network.md [hunk 4]
```diff
@@ -71,7 +75,7 @@ If you are using the Create Virtual Container Host wizard, stay on the Configure
 
 ## Example `vic-machine` Command <a id="example"></a>
 
-This example `vic-machine create` command deploys a VCH that designates an existing port group named `vch1-bridge` as the bridge network. It specifies IP addresses in the range 192.168.100.0/16 for use by user-defined bridge networks.
+This example `vic-machine create` command deploys a VCH that designates an existing interface named `vch1-bridge` as the bridge network. It specifies IP addresses in the range 192.168.100.0/16 for use by user-defined bridge networks.
```
Comment (yuyangbj) [inline docs/user_doc/vic_vsphere_admin/bridge_network.md]: An existing network is better?

File: docs/user_doc/vic_vsphere_admin/container_networks.md [hunk 5]
```diff
@@ -15,9 +15,14 @@ Running `docker network ls` lists the container networks, and container develope
 - [What to Do Next](#whatnext)
 - [Example `vic-machine` Command](#example)
 
+**IMPORTANT**: 
+
+- For information about VCH networking requirements, see [Networking Requirements for VCH Deployment](network_reqs.md#vchnetworkreqs).
+- If you use NSX-T Data Center logical switches, it is not mandatory for T1 or T0 routers to be present. For container networks, if you need to access container VMs externally, you might need a T1 router to bridge overlay and underlay networks. If you only want to validate connections to container VMs via the container network, you do not require T1 and T0 routers.
```
Comment (yuyangbj) [inline docs/user_doc/vic_vsphere_admin/container_networks.md]: you might need a T1..... this is just an example! So maybe we do not need to add it? We need to leave the opening to customers?

File: docs/user_doc/vic_vsphere_admin/interop.md [hunk 6]
```diff
@@ -1,67 +1,84 @@
...
 
-- If you restart a VCH endpoint VM, it comes back up in the same state that it was in when it shut down. 
-- If you use DHCP on the client network, the IP address of the VCH endpoint VM might change after a restart. Use `vic-machine inspect` to obtain the new IP address.
-- Do not manually delete a VCH resource pool, the VCH endpoint VM, or container VMs. Always use the vSphere Integrated Containers plug-in for the vSphere Client or `vic-machine delete` to delete VCHs. Always use Docker commands or the vSphere Integrated Containers Management Portal to perform operations on container VMs.
-- Manually restarting container VMs can result in incorrect end-times for container operations. Do not manually restart container VMs. Always use Docker commands or the vSphere Integrated Containers Management Portal to perform operations on container VMs.
+- You can deploy VCHs to NSX Data Center for vSphere networks if those networks are configured to provide distributed port groups.
+- You can deploy VCHs that use NSX-T Data Center logical switches instead of port groups. vSphere Integrated Containers supports NSX-T Data Center version 2.3.
```
Comment (yuyangbj) [inline docs/user_doc/vic_vsphere_admin/interop.md]: Actually, we only use the simple feature for NSXT in VIC, we can support 2.0, 2.2 and 2.3.

File: docs/user_doc/vic_vsphere_admin/bridge_network.md [hunk 7]
```diff
@@ -10,40 +10,44 @@ In Docker terminology, the bridge network on a VCH corresponds to the default br
 - [What to Do Next](#whatnext)
 - [Example `vic-machine` Command](#example)
 
+**IMPORTANT**: 
+
+- For information about VCH networking requirements, see [Networking Requirements for VCH Deployment](network_reqs.md#vchnetworkreqs).
+- If you use NSX-T Data Center logical switches, it is not mandatory for T1 or T0 routers to be present. The bridge network does not use layer 3, so does not require T1 and T0 routers.
```
[the remark this answers is not in the document]
Comment (stuclem) [reply docs/user_doc/vic_vsphere_admin/bridge_network.md]: These bullets are embedded under the IMPORTANT heading, so we don't need NOTE here.

File: docs/user_doc/vic_vsphere_admin/bridge_network.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (stuclem) [reply docs/user_doc/vic_vsphere_admin/bridge_network.md]: The official names for NSX-V and NSX-T are now as follows:

- NSX Datacenter for vSphere (formerly NSX-V)
- NSX-T Data Center (formerly just NSX-T)

Comment (stuclem) [reply docs/user_doc/vic_vsphere_admin/bridge_network.md]: Reworded to "Do not specify the same port group or logical switch as the bridge network for multiple VCHs."

File: docs/user_doc/vic_vsphere_admin/bridge_network.md (hunk 2 above)
Comment (stuclem) [reply docs/user_doc/vic_vsphere_admin/bridge_network.md]: Reworded to "Do not use the bridge network for any of the other VCH networking options."

File: docs/user_doc/vic_vsphere_admin/bridge_network.md (hunk 3 above)
Comment (stuclem) [reply docs/user_doc/vic_vsphere_admin/bridge_network.md]: Fixed

File: docs/user_doc/vic_vsphere_admin/bridge_network.md (hunk 4 above)
Comment (stuclem) [reply docs/user_doc/vic_vsphere_admin/bridge_network.md]: Fixed.

File: docs/user_doc/vic_vsphere_admin/container_networks.md (hunk 5 above)
Comment (stuclem) [reply docs/user_doc/vic_vsphere_admin/container_networks.md]: Reduced this bullet to "If you use NSX-T Data Center logical switches, you might need a T1 router to bridge overlay and underlay networks." Is this OK or should I delete the whole bullet?

File: docs/user_doc/vic_vsphere_admin/interop.md (hunk 6 above)
Comment (stuclem) [reply docs/user_doc/vic_vsphere_admin/interop.md]: Fixed. What about 2.1?

File: docs/user_doc/vic_vsphere_admin/mgmt_network.md [hunk 8]
```diff
@@ -10,22 +10,23 @@ The management network is the network on which the VCH endpoint VM connects to v
...
+- If you use a dedicated interface for the management network, the port group or logical switch must exist before you create the VCH.
 - Because the management network provides access to your vSphere environment, and because container VMs use this network to communicate with the VCH, always use a secure network for the management network.
 - Container VMs communicate with the VCH endpoint VM over the management network when an interactive shell is required. While the communication is encrypted, the public keys are not validated, which leaves scope for man-in-the-middle attacks. This connection is only used when the interactive console is enabled (`stdin`/`out`/`err`), and not for any other purpose. 
 - Ideally, use separate networks for the management network and container networks. 
-- You can use the same port group as the management network for multiple VCHs.
+- You can use the same interface as the management network for multiple VCHs.
```
[the remark this answers is not in the document]
Comment (stuclem) [reply docs/user_doc/vic_vsphere_admin/mgmt_network.md]: Fixed

File: docs/user_doc/vic_vsphere_admin/network_reqs.md [hunk 9]
```diff
@@ -39,26 +38,94 @@ See also [Docker container networking](https://docs.docker.com/engine/userguide/
...
-- Each VCH requires access to at least one network, for use as the public network. You can share this network between multiple VCHs. The public network does not need to be trusted.
+- [NSX-T Data Center Installation Guide](https://docs.vmware.com/en/VMware-NSX-T-Data-Center/2.3/com.vmware.nsxt.install.doc/GUID-3E0C4CEC-D593-4395-84C4-150CD6285963.html)
+- [Create a Logical Switch
+](https://docs.vmware.com/en/VMware-NSX-T-Data-Center/2.3/com.vmware.nsxt.admin.doc/GUID-23194F9A-416A-40EA-B9F7-346B391C3EF8.html) 
 
-The following network requirements apply to the deployment of VCHs to vCenter Server: 
+**NOTE**: If you use NSX-T Data Center logical switches, it is not mandatory for T1 or T0 routers to be present. The bridge network  does not use layer 3, so does not require T1 and T0 routers. For container networks, if you need to access container VMs externally, you might need a T1 router to bridge overlay and underlay networks. If you only want to validate connections to container VMs via the container network, you do not require T1 and T0 routers.
```
[the remark this answers is not in the document]
Comment (stuclem) [reply docs/user_doc/vic_vsphere_admin/network_reqs.md]: Reworded to "For container networks, you might need a T1 router to bridge overlay and underlay networks." Remove completely or is this OK?

**Comment by stuclem** [comment]
Hi @yuyangbj I have addressed your comments in https://github.com/vmware/vic-product/pull/2264/commits/26977cd08f05ed43096f30f3373ee4f52fa9acd5. Can you please take another look?

File: docs/user_doc/vic_vsphere_admin/bridge_network.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (yuyangbj) [reply docs/user_doc/vic_vsphere_admin/bridge_network.md]: Thanks for reminding!

Comment (yuyangbj) [reply docs/user_doc/vic_vsphere_admin/bridge_network.md]: Cool!

File: docs/user_doc/vic_vsphere_admin/interop.md (hunk 6 above)
Comment (yuyangbj) [reply docs/user_doc/vic_vsphere_admin/interop.md]: Agree.

File: docs/user_doc/vic_vsphere_admin/network_reqs.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (yuyangbj) [reply docs/user_doc/vic_vsphere_admin/network_reqs.md]: We can leave it as an example.

File: docs/user_doc/vic_vsphere_admin/interop.md (hunk 6 above)
Comment (stuclem) [reply docs/user_doc/vic_vsphere_admin/interop.md]: Added 2.1

File: docs/user_doc/vic_vsphere_admin/network_reqs.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (stuclem) [reply docs/user_doc/vic_vsphere_admin/network_reqs.md]: Updated to "For container networks, you might need a T1 router, for example to bridge overlay and underlay networks."

**Comment by stuclem** [comment]
Removed the statement about not being able to use NSX-T/V in non-VC environments, following confirmation from @lgayatri and @jitinkumar2018 that you can indeed use them with standalone ESXi hosts. 

@yuyangbj is this OK to merge now? Thanks!

**Comment by stuclem** [comment]
Thanks @yuyangbj !

**Pull request merged**